lib/Dialect/Torch/Transforms/GlobalizeObjectGraph.cpp

```diff
@@ -142,11 +143,36 @@ class ObjectGraphInfo {
       if (slotIt == nameToSlot.end())
         op->emitError() << "Reference to non-existing module slot " << slotName
                         << "in " << moduleType.getClassName();
-      usedSlots.insert(slotIt->getValue());
+      slotUses[slotIt->getValue()].insert(op);
     });
     return success();
   }
 
+  LogicalResult checkUsesForValueSemantics(Value root) {
+    for (auto &use : root.getUses()) {
+      // For non-slot operations, simply trace the value's uses.
+      if (!isa<SlotOp>(use.getOwner()))
+        return checkUsesForValueSemantics(use.get());
```

Comment:
isn't `use.get() == root` always? Doesn't this code infinitely recurse?

Reply:
Oh dang, thanks for catching that.  I meant to say `use.getOwner()->getResults()`.  Fixed and updated the test to exercise this code path.